src/Receipts/ReceiptResponse.php

```diff
@@ -1,90 +1,110 @@
 <?php
 
-
 namespace Imdhemy\AppStore\Receipts;
 
-use Imdhemy\AppStore\Exceptions\InvalidReceiptException;
+use Imdhemy\AppStore\ValueObjects\LatestReceiptInfo;
 use Imdhemy\AppStore\ValueObjects\PendingRenewal;
 use Imdhemy\AppStore\ValueObjects\Receipt;
-use Imdhemy\AppStore\ValueObjects\ReceiptInfo;
 use Imdhemy\AppStore\ValueObjects\Status;
 
 /**
  * Class ReceiptResponse
  * @package Imdhemy\AppStore\Receipts
+ * @see https://developer.apple.com/documentation/appstorereceipts/responsebody
  */
 class ReceiptResponse
 {
+    public const ENV_SANDBOX = 'Sandbox';
+    public const ENV_PRODUCTION = 'Production';
+
     /**
-     * @var string
+     * The environment for which the receipt was generated.
+     * @var string|null
      */
     protected $environment;
 
     /**
+     * An indicator that an error occurred during the request.
      * @var bool|null
      */
     protected $isRetryable;
 
     /**
+     * The latest Base64 encoded app receipt.
+     * Only returned for receipts that contain auto-renewable subscriptions.
      * @var string|null
      */
     protected $latestReceipt;
 
     /**
-     * @var array|ReceiptInfo[]
+     * An array that contains all in-app purchase transactions.
+     * @var array|LatestReceiptInfo[]|null
      */
     protected $latestReceiptInfo;
```

Comment:
since this is a major refactoring, I'd strongly consider dropping everything below PHP8 and go with native union types

Reply:
even if the union types are not fully applicable for the typed array in that case

Reply:
@michavie 
This is not clear for me, Could you please add an example?

Reply:
since you're working on the major version with big refactorings, my idea was to require a minimum version of PHP8 to further simplify code with native union types, constructor prop promotion, and all of the other useful new.

union types in that case would have worked if LatestReceiptInfo[] wasn't a typed array which I missed when writing the initial comment but would cetrainly work in other places of the code.